File: C++/A/975A-aramicscript.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n;
    cin >> n;
    string new_word;
    set<string> words;
    while (n--)
    {
        cin >> new_word;
        if (new_word.size() == 1)
            words.insert(new_word);
        else
        {
            sort(new_word.begin(), new_word.end());
            set<char> seen;
            string result = "";
            for (char c : new_word)
                if (seen.insert(c).second)
                    result += c;
            words.insert(result);
        }
    }
    cout << words.size();
}

File: Python/A/975A-aramicscript.py
n = int(input())
new_words = list(input().split())
words = set()

for new_word in new_words:
    if len(new_word) == 1:
        words.add(new_word)
    else:
        new_word = sorted(new_word)
        seen = set()
        result = ""
        for c in new_word:
            if c not in seen:
                result += c
                seen.add(c)
        words.add(result)
print(len(words))
